blog/2023-06-16-inline-sdk.mdx

```diff
@@ -0,0 +1,1104 @@
+
+```
+npx @vertigis/inline-sdk generate VideoBand
+```
+
+:::important
+The `generate` command must be run from within the custom band project folder.
+:::
+
+### Add settings for the Inline Designer
+
+The `designer.ts` file provides the Inline designer with a settings schema to display configurable settings for a custom band. These settings can be any combination of:
+
+-   input
+-   checkbox
+-   radio
+-   dropdown
+
+The Sample Band has some examples of how to use these settings. The `src/bands/VideoBand/designer.ts` file will have a small function ready to go.
+
+```ts title="src/bands/VideoBand/designer.ts"
+import { InlineDesignerConfig } from "inline";
+
+export const getDesignerConfig = (): InlineDesignerConfig => {
+    const config: InlineDesignerConfig = [
+        {
+            title: "Video Band Settings",
+            fields: [],
+        },
+    ];
+
+    return config;
+};
+```
+
+The `InlineDesignerConfig` is a list of settings sections. Each section has a name and list of fields that can be configured by users in the Inline designer.
+
+Next, add an input field for the video URL.
+
+```ts title="src/bands/VideoBand/designer.ts"
+// highlight-next-line
+import { InlineDesignerConfig, InputDesignerConfig } from "inline";
+
+export const getDesignerConfig = () => {
+    const config: InlineDesignerConfig = [
+        {
+            title: "Video Band Settings",
+            fields: [],
+        },
+    ];
+
+    // highlight-start
+    const titleField: InputDesignerConfig = {
+        id: "url",
+        type: "input",
+        default: "",
+        label: "Video URL",
+        helpText:
+            "The URL of the video. This must be a valid URL to a video file.",
+    };
+
+    config[0].fields.push(titleField);
+    // highlight-end
+
+    return config;
+};
+```
+
+This pattern can be repeated to create any other settings that are necessary for the band.
+
+### Creating a Video Player React Hook
+
+Without a video player, this band will not do much. Let's create one using a custom React Hook.
+
+Create a new file the the `VideoBand` folder called `useVideoPlayer.ts`. In this file we will include all the tools we need to create an interactive video player in a custom band.
+
+Set up the hook as a function that takes a single parameter, `videoElement`. This is a reference to a video HTML element.
+
+```ts title="src/bands/VideoBand/useVideoPlayer.ts"
+const useVideoPlayer = (
+    videoElement: React.MutableRefObject<HTMLVideoElement>
+) => {};
+
+export default useVideoPlayer;
+```
+
+Add some state to track to current status of the player.
+
+```ts title="src/bands/VideoBand/useVideoPlayer.ts"
+// highlight-next-line
+import { useState } from "react";
+
+const useVideoPlayer = (
+    videoElement: React.MutableRefObject<HTMLVideoElement>
+) => {
+    // highlight-start
+    const [playerState, setPlayerState] = useState({
+        isPlaying: false,
+        progress: 0,
+        speed: 1,
+        isMuted,
```

Comment:
Do we need to initialize `isMuted: false`?

Reply:
Fixed